chartfx-chart/src/main/java/de/gsi/chart/utils/WriteFxImage.java

```diff
@@ -0,0 +1,244 @@
+package de.gsi.chart.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.util.Map;
+import java.util.zip.CRC32;
+import java.util.zip.Deflater;
+
+import javafx.scene.image.Image;
+import javafx.scene.image.PixelReader;
+
+import de.gsi.dataset.utils.ArrayCache;
+
+/**
+ * Writes a JavaFx Image into a ByteBuffer or file
+ * 
+ * possible improvements:
+ * - use multiple IDAT chunks to use fixed buffer size and limit memory footprint
+ * - implement filtering of lines before compression for smaller file sizes
+ * - Optionally add tEXT chunks for metadata (EXIF)
+ *
+ * @author Alexander Krimm
+ */
+public final class WriteFxImage {
+    protected static final int HEADER_SIZE = 8 + (12 + 13) + 12 + 12; // size of all the headers and other Metadata
+    protected static final String INTERNAL_ARRAY_CACHE_NAME = "WriteFxImage-internalArray";
+
+    /**
+     * private constructor for static utility class
+     */
+    private WriteFxImage() {
+    }
+
+    /**
+     * Encodes a JavaFx image as an RGB png image with maximum (lossless) compression 
+     *
+     * @param image The input image to be encoded
+     * @return a byte buffer with the encoded image
+     * @see <a href="https://tools.ietf.org/html/rfc2083">rfc2083</a>
+     */
+    public static ByteBuffer encode(final Image image) {
+        return encode(image, null, true, Deflater.BEST_COMPRESSION, null);
```

Comment:
I'd argue for using `BEST_SPEED` as the default compression level.

I haven't tried image data, but I did some tests for time series data and found that `BEST_COMPRESSION` resulted in a marginal ~5-10% smaller file size while taking 30-40x longer to compute.